File: api/src/main/java/com/yunzhi/schedule/entity/Dispatch.java
package com.yunzhi.schedule.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonView;
import io.swagger.annotations.ApiModelProperty;
import org.hibernate.annotations.SQLDelete;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@SQLDelete(sql = "update `dispatch` set deleted = 1 where id = ?")
public class Dispatch implements SoftDelete {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @ApiModelProperty("调度ID")
    @JsonView(IdJsonView.class)
    private Long id;

    @ManyToOne
    @ApiModelProperty("对应排课")
    @JsonIgnoreProperties({"dispatches"})
    @JsonView(ScheduleJsonView.class)
    private Schedule schedule = new Schedule();

    @ApiModelProperty("对应周")
    @JsonView(WeekJsonView.class)
    private Long week;
    @ApiModelProperty("对应天")
    @JsonView(DayJsonView.class)
    private Long day;
    @ApiModelProperty("对应节")
    @JsonView(LessonJsonView.class)
    private Long lesson;
    @ManyToMany
    @ApiModelProperty("对应教室")
    @JsonView(RoomsJsonView.class)
    private List<Room> rooms = new ArrayList<>();

    @ApiModelProperty("是否已删除")
    private Boolean deleted = false;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Schedule getSchedule() {
        return schedule;
    }

    public void setSchedule(Schedule schedule) {
        this.schedule = schedule;
    }

    public Long getWeek() {
        return week;
    }

    public void setWeek(Long week) {
        this.week = week;
    }

    public Long getDay() {
        return day;
    }

    public void setDay(Long day) {
        this.day = day;
    }

    public Long getLesson() {
        return lesson;
    }

    public void setLesson(Long lesson) {
        this.lesson = lesson;
    }

    public List<Room> getRooms() {
        return rooms;
    }

    public void setRooms(List<Room> rooms) {
        this.rooms = rooms;
    }

    @Override
    public Boolean getDeleted() {
        return this.deleted;
    }

    private void setDeleted(Boolean deleted) {
        this.deleted = deleted;
    }


    public interface IdJsonView {}
    public interface ScheduleJsonView {}
    public interface WeekJsonView {}
    public interface LessonJsonView {}
    public interface DayJsonView {}
    public interface RoomsJsonView {}
}

File: schedule/src/entity/dispatch.ts
import {Assert} from '../common/utils';
import {Schedule} from './schedule';
import {Room} from './room';

/**
 * 调度
 */
export class Dispatch {
  id: number;
  week: number | undefined;
  day: number | undefined;
  lesson: number | undefined;
  schedule = {} as Schedule;
  rooms = [] as Room[];

  constructor(data = {} as {
    id?: number,
    week?: number,
    day?: number,
    lesson?: number,
    schedule?: Schedule,
    rooms?: Room[],
  }) {
    this.id = (data.id as number);
    this.week = data.week!;
    this.day = data.day!;
    this.lesson = data.lesson!;
    this.schedule = data.schedule!;
    this.rooms = data.rooms!;
  }
  getSchedule(): Schedule {
    Assert.isDefined(this.schedule, '不满足获取schedule的前提条件');
    return this.schedule!;
  }
  getId(): number {
    Assert.isDefined(this.id, '不满足获取id的前提条件');
    return this.id!;
  }
  getWeek(): number {
    Assert.isDefined(this.week, '不满足获取week的前提条件');
    return this.week!;
  }
  getRooms(): Room[] {
    Assert.isDefined(this.rooms, '不满足获取rooms的前提条件');
    return this.rooms!;
  }
  getDay(): number {
    Assert.isDefined(this.day, '不满足获取day的前提条件');
    return this.day!;
  }
  getLesson(): number {
    Assert.isDefined(this.lesson, '不满足获取lesson的前提条件');
    return this.lesson!;
  }
}
